### ISSUE TITLE
False positive -Wundefined-reinterpret-cast with SSE types?

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#58760]
Hi!

We are getting the `-Wundefined-reinterpret-cast` warning from the following code:
```cpp
typedef double __v2df __attribute__((__vector_size__(16)));

int main()
{
    double src[2] __attribute__((__aligned__(16)));
    __v2df dst{};
    *reinterpret_cast<__v2df*>(&src) = dst;
}
```

```
<source>:7:6: warning: dereference of type '__v2df *' that was reinterpret_cast from type 'double (*)[2]' has undefined behavior [-Wundefined-reinterpret-cast]
    *reinterpret_cast<__v2df*>(&src) = dst;
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Repro: https://godbolt.org/z/fndoEq71P

Is this warning correct, or a false positive? Aren't both pointers pointing to `double`, therefore compatible types?
